### PR TITLE
Add flyway plugin to manage database schemas

### PR DESCRIPTION
**Pull request by Christewart**
Adds the [sbt-flyway plugin](https://davidmweber.github.io/flyway-sbt-docs/) to the node project. This will be using to manage our database schema changes in the future.

https://flywaydb.org/

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -261,6 +266,22 @@ lazy val doc = project
...
+  lazy val DB_HOST = "localhost"
+  lazy val DB_NAME = "nodedb"
+  lazy val unittest = List(
+    Test / flywayUrl := s"jdbc:sqlite:${System.getenv("HOME")}/.bitcoin-s/unittest/nodedb.sqlite",
+    Test / flywayLocations := List("nodedb/migration"),
+    Test / flywayUser := "nodedb",
+    Test / flywayPassword := "",
+    flywayUrl := s"jdbc:sqlite:${System.getenv("HOME")}/.bitcoin-s/unittest/nodedb.sqlite",
+    flywayUser := "nodedb",
+    flywayPassword := ""
```
Comment (torkelrogstad) [inline build.sbt:296]: Would it be an idea to do a env var here?

Comment (Christewart) [reply build.sbt:296]: We probably want to eventually, although with sqlite as the driver apparently it doesnt have users so it isn't really necessary now. I believe we could even remove the `nodedb` username setting.

File: build.sbt [hunk 2]
```diff
@@ -261,6 +266,22 @@ lazy val doc = project
...
+  )
+
+  unittest
```
Comment (nkohen) [inline build.sbt]: I assume in the future we'll have other options for this with configurable url etc?

Comment (nkohen) [reply build.sbt]: Currently migrations are only going to affect this one database and it looks like there are others?

File: build.sbt [hunk 3]
```diff
@@ -261,6 +266,74 @@ lazy val doc = project
...
+
+  //create directies if they DNE
+  dirs.foreach { d =>
+    val file = new File(d)
+    file.mkdirs()
+    val db = new File(d + DB_NAME)
+    db.createNewFile()
+  }
+
+  lazy val mainnet = List(
```
Comment (torkelrogstad) [inline build.sbt]: To cut down on the repetition we could do 

```scala
  def makeNetworkSettings(directory: File) = List(
    Test / flywayUrl := s"jdbc:sqlite:$directory$DB_NAME",
    Test / flywayLocations := List("nodedb/migration"),
    Test / flywayUser := "nodedb",
    Test / flywayPassword := "",
    flywayUrl := s"jdbc:sqlite:$directory$DB_NAME",
    flywayUser := "nodedb",
    flywayPassword := ""
  )

lazy val testnet3 = makeNetworkSettings(testnetDir)
// etc
```

**Pull request merged**